L.Control.Geonames.js

```diff
@@ -117,9 +119,15 @@ L.Control.Geonames = L.Control.extend({
                         this._map.removeLayer(this._marker);
                         this._marker = null;
                     }
-                    this._marker = L.marker([lat, lon]).addTo(this._map).bindPopup(this._getName(geoname));
-                    this._map.setView([lat, lon], this.options.zoomLevel || this._map.getMaxZoom(), false);
-                    this._marker.openPopup();
+		    if (this.options.markNames) {
+			this._marker = L.marker([lat, lon]);
+			this._marker.addTo(this._map).bindPopup(this._getName(geoname));
+			this._map.setView([lat, lon], zoomLevel, false);
```

Comment:
@mikemoraned can you please move this to before the `if` and then drop the `else` altogether?